apps/entity-service/src/entities/processors/media.processor.ts

```diff
@@ -0,0 +1,245 @@
+    const models: QueryModelType[] = [{ modelType: this.entityModelClass.LARAVEL_TYPE, subquery: [projectReport.id] }];
+
+    const projectReportWithDue = await ProjectReport.findOne({
+      where: { id: projectReport.id },
+      attributes: ["dueAt"]
+    });
+
+    const project = await Project.findOne({ where: { id: projectReport.projectId }, attributes: ["id"] });
+
+    const siteSubquery = Subquery.select(Site, "id").eq("projectId", project.id);
+    const nurserySubquery = Subquery.select(Nursery, "id").eq("projectId", project.id);
+
+    let siteReports = [];
+    if (projectReportWithDue.dueAt) {
+      siteReports = await SiteReport.findAll({
+        where: {
+          [Op.and]: [
+            { siteId: { [Op.in]: siteSubquery.literal } },
+            Sequelize.where(fn("MONTH", col("due_at")), projectReportWithDue.dueAt.getMonth() + 1),
+            Sequelize.where(fn("YEAR", col("due_at")), projectReportWithDue.dueAt.getFullYear())
+          ]
+        },
+        attributes: ["id"]
+      });
+    }
+
+    models.push({ modelType: SiteReport.LARAVEL_TYPE, subquery: siteReports.map(report => report.id) });
+
+    let nurseryReports = [];
+    if (projectReportWithDue.dueAt) {
+      nurseryReports = await NurseryReport.findAll({
+        where: {
+          [Op.and]: [
+            { nurseryId: { [Op.in]: nurserySubquery.literal } },
+            Sequelize.where(fn("MONTH", col("due_at")), projectReportWithDue.dueAt.getMonth() + 1),
+            Sequelize.where(fn("YEAR", col("due_at")), projectReportWithDue.dueAt.getFullYear())
+          ]
+        },
+        attributes: ["id"]
+      });
+    }
+    models.push({ modelType: NurseryReport.LARAVEL_TYPE, subquery: nurseryReports.map(report => report.id) });
+
+    return models;
+  }
+
+  private async getSiteReportModels(siteReport: SiteReport) {
+    return [{ modelType: this.entityModelClass.LARAVEL_TYPE, subquery: [siteReport.id] }];
+  }
+
+  private async buildQuery(baseEntity: EntityModel, query: MediaQueryDto) {
+    const userAssociations: Includeable = {
+      association: "createdByUser",
+      attributes: ["firstName", "lastName"]
+    };
+
+    const builder = await this.entitiesService.buildQuery(Media, query, [userAssociations]);
+
+    let models = [];
+    if (baseEntity instanceof Project) {
+      models = await this.getProjectModels(baseEntity);
+    } else if (baseEntity instanceof Site) {
+      models = await this.getSiteModels(baseEntity);
+    } else if (baseEntity instanceof Nursery) {
+      models = await this.getNurseryModels(baseEntity);
+    } else if (baseEntity instanceof ProjectReport) {
+      models = await this.getProjectReportModels(baseEntity);
+    } else if (baseEntity instanceof SiteReport) {
+      models = await this.getSiteReportModels(baseEntity);
+    } else if (baseEntity instanceof NurseryReport) {
+      models = [{ modelType: this.entityModelClass.LARAVEL_TYPE, subquery: [baseEntity.id] }];
```

Comment:
the `getSiteReportModels` method is doing this exact same thing. I'd either remove that method entirely, or rename it to something like `getBaseEntityModels()` and use it in both cases.

Reply:
changed